woocommerce/woocommerce-ios
# Order Creation - Order Sync: Adds Fees support

Closes: #6141 

# Why

This PR adds proper sync support to adding, updating, and removing fee lines.

# How

- Adds a new type `FeeInputTransformer` to take care of updating an order given the fee input.

- Make sure that we set the fee `name` to `nil` when trying to delete a fee line.

# Demo

https://user-images.githubusercontent.com/562080/157109469-e02e666f-4296-43e5-b463-2a30f2b19f0c.mov


# Testing Steps

- Set `orderCreationRemoteSynchronizer` feature flag to `true`
- Go to the new order screen.
- Add a fee & see the totals updated.
- Updates the fee & see the totals updated.
- Remove the fee & see the fee being removed.

---
